From 7b89e18c2f21e4b91b461007368df724198ac904 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Tue, 12 Mar 2024 03:13:33 +0100
Subject: [PATCH 1/2] fix(drm): Default to XRGB8888 framebuffer

The ARGB8888 framebuffer format for base canvas makes little sense
as the base canvas is unlikely to be transparent and require alpha.
Use XRGB8888 framebuffer format which is more widely supported by
DRM drivers as base plane pixel format.

This makes e.g. i.MX8M Nano work by default.

Upstream-Status: Submitted [https://github.com/lvgl/lvgl/pull/5851]
Signed-off-by: Marek Vasut <marex@denx.de>
---
This is identical fix to lv_drivers https://github.com/lvgl/lv_drivers/pull/282
---
 src/drivers/display/drm/lv_linux_drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/display/drm/lv_linux_drm.c b/src/drivers/display/drm/lv_linux_drm.c
index c706f9cb2..faa2212bf 100644
--- a/src/drivers/display/drm/lv_linux_drm.c
+++ b/src/drivers/display/drm/lv_linux_drm.c
@@ -24,7 +24,7 @@
  *      DEFINES
  *********************/
 #if LV_COLOR_DEPTH == 32
-    #define DRM_FOURCC DRM_FORMAT_ARGB8888
+    #define DRM_FOURCC DRM_FORMAT_XRGB8888
 #elif LV_COLOR_DEPTH == 16
     #define DRM_FOURCC DRM_FORMAT_RGB565
 #else
-- 
2.43.0

